pkg/api/pluginproxy/ds_proxy.go

```diff
@@ -320,9 +320,15 @@ func (proxy *DataSourceProxy) applyRoute(req *http.Request) {
 		SecureJsonData: proxy.ds.SecureJsonData.Decrypt(),
 	}
 
-	routeURL, err := url.Parse(proxy.route.Url)
+	interpolatedURL, err := interpolateString(proxy.route.Url, data)
```

Comment:
Probably not in scope for this PR, but shouldn't applyRoute return error?

Reply:
Don't think it can as it is using the Director function from [httputil.reverseproxy](https://golang.org/pkg/net/http/httputil/#ReverseProxy). Also see [Go issue](https://github.com/golang/go/issues/14329)

Reply:
yea, I struggled with this as well, the go http proxy director function's lack of error return makes good error handling in the reverse proxy almost impossible.